### PR TITLE
Respect `excludes` only on runtime classpath

### PR DESCRIPTION
**Pull request by Duhemm**
Previously, Pants would respect `excludes` on both the compilation and
runtime classpaths. It turns out that, with Pants, the `excludes` should
only apply to the runtime classpath: the compilation classpath should
still include all libraries, regardless of the `excludes`.

### CODE REVIEW

File: tests/slow/src/test/scala/tests/BloopPantsSuite.scala [hunk 1]
```diff
@@ -11,6 +11,8 @@ class BloopPantsSuite extends FastpassSuite {
                            |/pants.ini
                            |[GLOBAL]
                            |pants_version: 1.26.0.dev0
+                           |[resolver]
+                           |resolver: coursier
```
Comment (olafurpg) [inline tests/slow/src/test/scala/tests/BloopPantsSuite.scala:15]: TIL

File: fastpass/src/main/scala/scala/meta/internal/fastpass/pantsbuild/BloopPants.scala [hunk 2]
```diff
@@ -165,6 +165,7 @@ object BloopPants {
       s"--${noInternalSources}export-fastpass-sources",
       s"--${no3rdPartySources}export-fastpass-libraries-sources",
       s"--export-fastpass-output-file=$outputFile",
+      s"--export-fastpass-allow-global-excludes=False",
```
Comment (olafurpg) [inline fastpass/src/main/scala/scala/meta/internal/fastpass/pantsbuild/BloopPants.scala:168]: Is this flag necessary? There are no other clients using `export-fastpass` so I wonder if we can make this flag enabled by default 🤔

Comment (Duhemm) [reply fastpass/src/main/scala/scala/meta/internal/fastpass/pantsbuild/BloopPants.scala:168]: We can set this from `pants.ini` instead. I personally prefer to have the tool "configure itself", but I'm happy to change it if you think that'd be better.

File: tests/slow/src/test/scala/tests/BloopPantsSuite.scala (hunk 1 above)
Comment (Duhemm) [reply tests/slow/src/test/scala/tests/BloopPantsSuite.scala:15]: Yes, it was surprising to me as well. I don't know why Ivy ignores the flag...

**Pull request merged**